### PR TITLE
Add configurable image positioning to Figure component

### LINKED ISSUE #65: Figures should allow a top and bottom
Right now, the Figure component only allows displaying the image *above* the content.

Ideally, we would support some way to allow the user to control where it appears.

### PR DESCRIPTION
**Pull request by topherfangio**
## Context
The Figure component currently only supports displaying images above content, limiting flexibility in layout design. Users need the ability to control image positioning to better suit their content flow and design requirements.

## Related Issues
Fixes #65

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [x] Component update/addition
- [x] Test update/addition

## Description
This enhancement adds configurable image positioning to the Figure component:

Screenshot 2025-12-04 at 3 50 49 PM

### Key Changes:
- **New `position` parameter**: Accepts `"top"` (default) or `"bottom"` to control image placement
- **Backward compatibility**: Default behavior unchanged when no position specified
- **Input validation**: Raises clear `ArgumentError` for invalid position values
- **Comprehensive testing**: Full test coverage for all positioning scenarios
- **Enhanced documentation**: Updated examples and YARD documentation

### Implementation Details:
- Modified `FigureComponent#initialize` to accept and validate position parameter
- Updated `FigureComponent#call` to conditionally render image before or after content
- Added extensive unit tests covering default behavior, explicit positioning, and edge cases
- Enhanced demo examples with new "Figure with Bottom Position" showcase
- Updated E2E tests to verify positioning functionality works correctly

### Usage Examples:
```ruby
# Default: image above content
daisy_figure(src: "image.jpg") { "Caption" }

# Explicit top positioning
daisy_figure(src: "image.jpg", position: "top") { "Caption" }

# New: image below content
daisy_figure(src: "image.jpg", position: "bottom") { "Caption" }
```

### CODE REVIEW

File: app/components/daisy/data_display/figure_component.rb [hunk 1]
```diff
@@ -43,12 +53,27 @@ def before_render
...
   def call
     part(:component) do
-      if @src
-        concat(part(:image))
+      if @position == "bottom"
+        # Show content first, then image
+        concat(content)
+        if @src
+          concat(part(:image))
+        end
```
Comment (topherfangio) [inline app/components/daisy/data_display/figure_component.rb]: Should use the following:


```suggestion
        concat(part(:image)) if @src
```

**Review by topherfangio — COMMENTED** [review]
Fixes for Windsurf
